task/push-build-file/0.1/push-build-file.yaml

```diff
@@ -0,0 +1,97 @@
+  params:
+  - name: IMAGE
+    description: The built binary image. The build file is pushed to the same image repository alongside.
+    type: string
+  - name: IMAGE_DIGEST
+    description: The built binary image digest, which is used to construct the tag of build file image.
+    type: string
+  - name: BUILD_FILE
+    description: Path to the build file.
+    type: string
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: Path to the directory to use as context.
+    type: string
+    default: .
+  - name: TAG_SUFFIX
+    description: Suffix of the build file image tag.
+    type: string
+    default: .dockerfile
+  - name: ARTIFACT_TYPE
+    description: Artifact type of the build file image.
+    type: string
+    default: application/vnd.konflux.dockerfile
+  results:
+  - name: IMAGE_URL
+    description: Digest-pinned image URL.
+  steps:
+  - name: push
+    image: quay.io/konflux-ci/oras@sha256:5d0a8a5535fcc4ba467264cacbdeab2fb8662a538a61cb7fc8b3155e3f20fa39
+    workingDir: $(workspaces.workspace.path)
+    env:
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: IMAGE_DIGEST
+      value: $(params.IMAGE_DIGEST)
+    - name: TAG_SUFFIX
+      value: $(params.TAG_SUFFIX)
+    - name: BUILD_FILE
+      value: $(params.BUILD_FILE)
+    - name: CONTEXT
+      value: $(params.CONTEXT)
+    - name: ARTIFACT_TYPE
+      value: $(params.ARTIFACT_TYPE)
+    - name: IMAGE_URL_RESULT
+      value: $(results.IMAGE_URL.path)
+    script: |
+      set -o pipefail
+
+      # Same discovery logic used in buildah task
+      SOURCE_CODE_DIR=source
+      if [ -e "$SOURCE_CODE_DIR/$CONTEXT/$BUILD_FILE" ]; then
+        build_file_path="$(pwd)/$SOURCE_CODE_DIR/$CONTEXT/$BUILD_FILE"
+      elif [ -e "$SOURCE_CODE_DIR/$BUILD_FILE" ]; then
+        build_file_path="$(pwd)/$SOURCE_CODE_DIR/$BUILD_FILE"
+      elif echo "$BUILD_FILE" | grep -q "^https\?://"; then
+        echo "Fetch build file from $BUILD_FILE"
+        build_file_path=$(mktemp --suffix=-build-file)
+        http_code=$(curl -s -L -w "%{http_code}" --output "$build_file_path" "$BUILD_FILE")
+        if [ $http_code != 200 ]; then
+          echo "No build file is fetched. Server responds $http_code"
+          exit 1
+        fi
+      else
+        echo "Cannot find build file $BUILD_FILE"
+        exit 1
+      fi
+
+      echo "Selecting auth for $IMAGE"
+      auth_json=$(mktemp)
+      select-oci-auth $IMAGE >"$auth_json"
+
+      build_file_image=${IMAGE%:*}:${IMAGE_DIGEST/:/-}${TAG_SUFFIX}
+
+      cd "$(dirname $build_file_path)"
+      retry oras push --no-tty \
+        --format json \
+        --registry-config "$auth_json" \
+        --artifact-type "$ARTIFACT_TYPE" \
+        "$build_file_image" "$(basename $build_file_path)" \
+        | yq '.reference' >"$IMAGE_URL_RESULT"
```

Comment:
Doesn't this cause an extra `\n` at the end of the image url result?

Reply:
Nice catch. Updated.